### PR TITLE
Add 3.7 typehints, bring example up to 3.x

### PR DESCRIPTION
**Pull request by MattIPv4**

### CODE REVIEW

File: pyudmx/pyudmx.py [hunk 1]
```diff
@@ -75,7 +77,8 @@ def close(self):
             usb.util.dispose_resources(self._dev)
             self._dev = None
 
-    def _send_control_message(self, cmd, value_or_length=1, channel=1, data_or_length=1):
+    def _send_control_message(self, cmd: Union[1, 2], value_or_length: int = 1, channel: int = 1,
```
Comment (dhocker) [inline pyudmx/pyudmx.py]: Arguments to Union must be a type not a value. This should be cmd: int.

File: pyudmx/pyudmx.py [hunk 2]
```diff
@@ -75,7 +77,8 @@ def close(self):
...
+                              data_or_length: Union[1, bytearray] = 1) -> int:
```
Comment (dhocker) [inline pyudmx/pyudmx.py]: Same issue with Union here. This should be data_or_length: Union[int, bytearray] = 1

**Comment by dhocker** [comment]
I tested this on Python 3.6.5 (on a Raspberry Pi) and here is what I got.

(athomedmx3) pi@raspberrypi-dev:~/rpi/udmx-pyusb $ python example.py 
Traceback (most recent call last):
  File "example.py", line 22, in <module>
    from pyudmx import pyudmx
  File "/home/pi/rpi/udmx-pyusb/pyudmx/pyudmx.py", line 31, in <module>
    class uDMXDevice:
  File "/home/pi/rpi/udmx-pyusb/pyudmx/pyudmx.py", line 81, in uDMXDevice
    data_or_length: Union[1, bytearray] = 1) -> int:
  File "/usr/local/lib/python3.6/typing.py", line 682, in inner
    return func(*args, **kwds)
  File "/usr/local/lib/python3.6/typing.py", line 800, in __getitem__
    parameters = tuple(_type_check(p, msg) for p in parameters)
  File "/usr/local/lib/python3.6/typing.py", line 800, in <genexpr>
    parameters = tuple(_type_check(p, msg) for p in parameters)
  File "/usr/local/lib/python3.6/typing.py", line 374, in _type_check
    raise TypeError(msg + " Got %.100r." % (arg,))
TypeError: Union[arg, ...]: each arg must be a type. Got 1.

After making the changes to Union that I identified above, it worked. Maybe this works in Python 3.7 but it doesn't work in 3.6.5. However, the Python 3.7 doc says the arguments to Union must be types. That means that PyCharm probably has a suggestion bug (and I use it, too).

**Comment by MattIPv4** [comment]
Yeah I'll play further and talk to the Pyc team. This should be good to go now though.

**Comment by dhocker** [comment]
Merged commit 0b58860c14fab9052b422bf12f421d6217458dd0

**Pull request merged**